README.md

```diff
@@ -5,6 +5,27 @@ the
 user like authenticating the users, issuing the access/refresh tokens, api keys, validating the permissions for the
 users.
 
+## System flow diagram
+
+Below diagram depicts the system flow for Authentication service
+
+```mermaid
+graph LR;
+    A(Client) --->|Authenticate| B(Gateway) 
+    B -->|Authenticate| C(Auth Service) 
+    C -->|Authenticate| D(Keycloak Service)
+    C -->|Authorize| E(TDEI Database)
+```
+
+- Auth service is responsible for Authenticating and Authorizing the HTTP requests. Also, verifies the access token,
+  api-key and generates the secret token
+  for intra micro-service communication.
+- Gateway service is first point of contact for clients. Every request to gateway is Authenticated & Authorized against
+  the Auth service.
+- For Authentication, Auth service will interact with keycloak service for verification.
+    - Successful authentication auth servive will return with access_token & refresh_token.
+- For Authorization, Auth service will interact directly with TDEI database.

```

Comment:
What does this TDEI database have? What is stored in it? Why does it interact direct?